src/classes/traveler.js

```diff
@@ -0,0 +1,46 @@
+import User from './user';
+
+class Traveler extends User {
+  constructor(travelers, trips, destinations, traveler, today) {
+    super(travelers, trips, destinations, today)
+    this.id = traveler.id;
+    this.name = traveler.name;
+    this.travelerType = traveler.travelerType;
+  };
+
+  getMyTrips() {
+    return this.destinations.reduce((trips, destination) => {
+      this.trips.find(trip => (this.id === trip.userID && destination.id === trip.destinationID) && trips.push(destination));
+      return trips;
+    }, []);
+
+  };
+
+  getYearlyTripExpenses() {
+    let travelerTrips = this.trips.filter(trip => trip.userID === this.id);
+    let tripDestinations = this.getMyTrips();
+    return travelerTrips.map(trip => {
+      let finalObj = { cost: 0, date: trip.date }
+        tripDestinations.forEach(destination => {
+          finalObj.cost += (trip.duration * destination.estimatedLodgingCostPerDay) + (trip.travelers * destination.estimatedFlightCostPerPerson)
+      });
+      return finalObj
+    })
+  };
+
+  // should get the last year from todays date BUT!! ... It's a leap year -_-
+  calculateyearlyTripExpenses() {
+    let yearlyTripExpenses = this.getYearlyTripExpenses().reduce((acc, el) => {
```

Comment:
It's a little hard to understand what the reduce is doing with the params labeled 'acc' and 'el'.